pkg/sources/chunker_test.go

```diff
@@ -84,3 +66,116 @@ func BenchmarkChunker(b *testing.B) {
 		}
 	}
 }
+
+func TestNewChunkedReader(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      string
+		chunkSize  int
+		peekSize   int
+		wantChunks []string
+		wantErr    bool
+	}{
+		{
+			name:       "Smaller data than default chunkSize and peekSize",
+			input:      "example input",
+			chunkSize:  ChunkSize,
+			peekSize:   PeekSize,
+			wantChunks: []string{"example input"},
+			wantErr:    false,
+		},
+		{
+			name:       "Reader with no data",
+			input:      "",
+			chunkSize:  1024,
+			peekSize:   512,
+			wantChunks: []string{},
+			wantErr:    false,
+		},
+		{
+			name:       "Smaller data than chunkSize and peekSize",
+			input:      "small data",
+			chunkSize:  1024,
+			peekSize:   512,
+			wantChunks: []string{"small data"},
+			wantErr:    false,
+		},
+		{
+			name:       "Equal to chunkSize",
+			input:      strings.Repeat("a", 1024),
+			chunkSize:  1024,
+			peekSize:   512,
+			wantChunks: []string{strings.Repeat("a", 1024)},
+			wantErr:    false,
+		},
+		{
+			name:       "Equal to chunkSize + peekSize",
+			input:      strings.Repeat("a", 1536),
+			chunkSize:  1024,
+			peekSize:   512,
+			wantChunks: []string{strings.Repeat("a", 1024), strings.Repeat("a", 512)},
+			wantErr:    false,
+		},
+		{
+			name:       "EOF during peeking",
+			input:      strings.Repeat("a", 1300),
+			chunkSize:  1024,
+			peekSize:   512,
+			wantChunks: []string{strings.Repeat("a", 1024), strings.Repeat("a", 276)},
+			wantErr:    false,
+		},
+		{
+			name:       "EOF during reading",
+			input:      strings.Repeat("a", 512),
+			chunkSize:  1024,
+			peekSize:   512,
+			wantChunks: []string{strings.Repeat("a", 512)},
+			wantErr:    false,
+		},
+		{
+			name:       "Equal to totalSize",
+			input:      strings.Repeat("a", 2048),
+			chunkSize:  1024,
+			peekSize:   1024,
+			wantChunks: []string{strings.Repeat("a", 1024), strings.Repeat("a", 1024)},
+			wantErr:    false,
+		},
```

Comment:
Shouldn't the first chunk have 2048 bytes (chunk size + peek size)?

Reply:
fixed